tests/func/test_data_cloud.py

```diff
@@ -77,12 +77,20 @@ class TestDataCloudBase(TestDvc):
     def _get_cloud_class(self):
         return None
 
-    def _should_test(self):
+    @staticmethod
+    def should_test():
         return False
 
-    def _get_url(self):
+    @staticmethod
+    def get_url():
         return ""
 
+    def _should_test(self):
+        return self.should_test.__get__(None, object)()
+
+    def _get_url(self):
+        return self.get_url.__get__(None, object)()
+
```

Comment:
This looks a bit weird. Maybe we could simply use `self.should_test` and `get_url` directly?

Reply:
`should_test()` and `get_url()` are `staticmethod`, so, we cannot do `self.should_test()`, and I wanted to use it (i.e. `S3`, `GCP` etc. helper classes) as a mixin on inherited classes downward. I'm still looking into other options.

Reply:
Ah, it worked. Don't know what I changed. I was getting `requires 0 arguments but given 1` when doing `self.should_test()`. :confused:

Reply:
Found it. Before, `should_test` was a lambda function, but, as I've changed it to `staticmethod`, it works. :sweat_smile: